diesel/src/sqlite/types/floats.rs

```diff
@@ -0,0 +1,22 @@
+use byteorder::{NativeEndian, WriteBytesExt};
+use serialize::{self, IsNull, Output, ToSql};
+use sql_types::{Double, Float};
+use sqlite::Sqlite;
+use std::error::Error;
+use std::io::Write;
+
+impl ToSql<Double, Sqlite> for f64 {
```

Comment:
Is there a reason this can't be:

```rust
impl<DB, ByteOrder> ToSql<Double, DB> for f64
where
    DB: Backend + for<'a> BinaryRawValue<'a, ByteOrder = ByteOrder>,
{
    fn to_sql<W: Write>(&self, out: &mut Output<W, DB>) -> serialize::Result {
        out.write_f64::<ByteOrder>(*self)
            .map(|_| IsNull::No)
            .map_err(|e| Box::new(e) as Box<dyn Error + Send + Sync>)
    }
}
```

It seems like that impl would eliminate most of the fallout from this.

Reply:
The problem here is that `Sqlite` not implements `BinaryRawValue`, but uses `SqliteValue` (which is an Enum) as raw value representation.